lib/osf-components/addon/components/expandable-preview/component.ts

```diff
@@ -0,0 +1,64 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+interface Args {
+    height?: number;
+    open?: boolean;
+}
+
+export default class ExpandablePreview extends Component<Args> {
+    @tracked thresholdHeight: number;
+    @tracked collapsed: boolean;
+    @tracked collapsible = false;
+    @tracked wrapperElement?: HTMLElement;
+
+    @action
+    didInsert(element: HTMLElement) {
+        this.wrapperElement = element;
+        this.checkThreshold(element);
+    }
+
+    @action
+    checkThreshold(element: HTMLElement) {
+        if (!this.wrapperElement) {
+            this.wrapperElement = element;
+        }
+        if (this.thresholdHeight < element.clientHeight) {
+            this.collapsible = true;
+            if (this.collapsed) {
+                this.collapseWrapper();
+            }
+        }
+    }
+
+    @action
+    toggleCollapse() {
+        this.collapsed = !this.collapsed;
+        if (this.collapsed) {
+            this.collapseWrapper();
+        } else {
+            this.expandWrapper();
+        }
+    }
+
+    @action
+    collapseWrapper() {
+        if (this.wrapperElement) {
+            this.wrapperElement.style.maxHeight = this.thresholdHeight + 'px';
+        }
+    }
+
+    @action
+    expandWrapper() {
+        if (this.wrapperElement) {
+            this.wrapperElement.style.maxHeight = '';
+        }
+    }
+
+    constructor(owner: unknown, args: Args) {
+        super(owner, args);
+        this.thresholdHeight = this.args.height || 200;
```

Comment:
I wanted to do something cool/clever here, by using ember-css-module's `@value` [documented here](https://github.com/salsify/ember-css-modules#values) and dynamically set that value to be the `thresholdHeight` defined here, but I couldn't make that work. Ideas/suggestions would be much appreciated!